lib/engine/game/g_1880/step/sell_company.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1880
+      module Step
+        class SellCompany < Engine::Step::Base
+          ACTIONS = %w[sell_company pass].freeze
+          def actions(entity)
+            return [] unless entity.player?
+            return [] unless can_sell?
+            return [] if sellable_companies(entity).empty? || @round.already_passed
+
+            ACTIONS
+          end
+
+          def auto_actions(entity)
+            return [Engine::Action::Pass.new(entity)] if sellable_companies(entity).empty? || @round.already_passed
+
+            company = sellable_companies(entity).first
+            price = sell_price(company)
+            return [Engine::Action::SellCompany.new(entity, company: company, price: price)] if @game.phase.name == 'B2'
+
+            []
+          end
+
+          def round_state
+            super.merge(
+            { already_passed: false }
+          )
+          end
+
+          def sellable_companies(entity)
+            entity.companies.select { |c| c.all_abilities.any? { |a| a.type == :sell_company } }
+          end
+
+          def can_sell?
+            @game.class::P0_AWARD.key?(@game.phase.name) && !@game.p0.closed?
+          end
+
+          def description
+            'Claim Woosong Award'
+          end
+
+          def blocks?
+            !sellable_companies(current_entity).empty?
+          end
+
+          def hide_corporations?
+            true
+          end
+
+          def log_skip(entity); end
+
+          def sell_price(_company)
+            @game.class::P0_AWARD[@game.phase.name]
+          end
+
+          def button_text
+            "Claim #{@game.format_currency(sell_price(nil))} cash award"
+          end
+
+          def process_sell_company(action)
+            company = action.company
+            player = action.entity
+            price = action.price
+            raise GameError, "#{player.name} doesn't own #{company.name}" if player != company.owner
+
+            player.companies.delete(company)
+            company.close!
+            @game.bank.spend(price, player) if price.positive?
+            @log << "#{player.name} claims #{@game.format_currency(price)} award for #{company.name}"
+          end
```

Comment:
this code is very similar to buy_sell_par_shares parring choice. Should I make a common module and extract this method?